File: sdk/typescript/src/compliance.ts
/**
 * TypeScript SDK Client for Enterprise AI Compliance Platform (Phase 5.27).
 */

export class ComplianceClient {
  private client: any;

  constructor(client: any) {
    this.client = client;
  }

  async adoptFramework(frameworkType: string, name: string, description: string = 'Framework'): Promise<any> {
    return this.client.request('POST', '/v1/compliance/frameworks', {
      framework_type: frameworkType,
      name,
      description,
    });
  }

  async listFrameworks(): Promise<any[]> {
    return this.client.request('GET', '/v1/compliance/frameworks');
  }

  async registerControl(code: string, name: string, description: string, controlType: string = 'PREVENTIVE', category: string = 'SECURITY'): Promise<any> {
    return this.client.request('POST', '/v1/compliance/controls', {
      code,
      name,
      description,
      control_type: controlType,
      category,
    });
  }

  async listControls(): Promise<any[]> {
    return this.client.request('GET', '/v1/compliance/controls');
  }

  async collectEvidence(idempotencyKey: string, subjectType: string, subjectId: string, requiredTypes: string[]): Promise<any> {
    return this.client.request('POST', '/v1/compliance/evidence/collect', {
      idempotency_key: idempotencyKey,
      subject_type: subjectType,
      subject_id: subjectId,
      required_evidence_types: requiredTypes,
    });
  }

  async runAssessment(frameworkId: string, subjectId: string = 'global'): Promise<any> {
    return this.client.request('POST', '/v1/compliance/assessments', {
      framework_id: frameworkId,
      subject_id: subjectId,
    });
  }

  async getPosture(): Promise<any> {
    return this.client.request('GET', '/v1/compliance/posture');
  }

  async generateAssuranceReport(frameworkId: string, conclusion: string = 'ASSURED'): Promise<any> {
    return this.client.request('POST', '/v1/compliance/assurance', {
      framework_id: frameworkId,
      conclusion,
    });
  }

  async createAuditPackage(frameworkId: string, evidenceBundleRef: string, assuranceReportRef: string): Promise<any> {
    return this.client.request('POST', '/v1/compliance/audit-packages', {
      framework_id: frameworkId,
      evidence_bundle_reference: evidenceBundleRef,
      assurance_report_reference: assuranceReportRef,
    });
  }

  async getAnalytics(): Promise<any> {
    return this.client.request('GET', '/v1/compliance/analytics');
  }
}

File: sdk/go/compliance.go
package llmengine

import (
	"context"
)

type ComplianceClient struct {
	client *Client
}

func NewComplianceClient(client *Client) *ComplianceClient {
	return &ComplianceClient{client: client}
}

func (c *ComplianceClient) AdoptFramework(ctx context.Context, frameworkType, name, description string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"framework_type": frameworkType,
		"name":           name,
		"description":    description,
	}
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/compliance/frameworks", payload, &res)
	return res, err
}

func (c *ComplianceClient) ListFrameworks(ctx context.Context) ([]map[string]interface{}, error) {
	var res []map[string]interface{}
	err := c.client.Get(ctx, "/v1/compliance/frameworks", &res)
	return res, err
}

func (c *ComplianceClient) RegisterControl(ctx context.Context, code, name, description, controlType, category string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"code":         code,
		"name":         name,
		"description":  description,
		"control_type": controlType,
		"category":     category,
	}
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/compliance/controls", payload, &res)
	return res, err
}

func (c *ComplianceClient) GetPosture(ctx context.Context) (map[string]interface{}, error) {
	var res map[string]interface{}
	err := c.client.Get(ctx, "/v1/compliance/posture", &res)
	return res, err
}
